### PR TITLE
feat(ORM): add orm podResources server

### PR DESCRIPTION
**Pull request by WangZzzhe**
#### What type of PR is this?
Enhancements

#### What this PR does / why we need it:
Add v1 podResources server for ORM. 
Topology allocated and allocatable resources from registered QRM plugins can be requested through [List] and [GetAllocatableResources] api.
ORM also request device resource from kubelet and merge datas to podResources response, it can be configured with katalyst-agent custom args.

#### Which issue(s) this PR fixes:
https://github.com/kubewharf/katalyst-core/issues/430

image

### CODE REVIEW

**Comment by waynepeking348** [comment]
need to rebase before it can be merged @WangZzzhe @caohe

**Comment by WangZzzhe** [comment]
> need to rebase before it can be merged @WangZzzhe @caohe

Done.

File: pkg/agent/resourcemanager/fetcher/kubelet/kubeletplugin.go [hunk 1]
```diff
@@ -237,9 +240,19 @@ func (p *kubeletPlugin) getNumaInfo() ([]info.Node, error) {
...
 func (p *kubeletPlugin) getTopologyPolicyReportContent(ctx context.Context) (*v1alpha1.ReportContent, error) {
-	topologyPolicy, err := p.topologyStatusAdapter.GetTopologyPolicy(ctx)
-	if err != nil {
-		return nil, errors.Wrap(err, "get topology policy from adapter failed")
+	var (
+		topologyPolicy nodev1alpha1.TopologyPolicy
+		err            error
+	)
+
+	if p.ormEnabled() {
```
Comment (waynepeking348) [inline pkg/agent/resourcemanager/fetcher/kubelet/kubeletplugin.go:248]: will this be necessary to have?

**Comment by waynepeking348** [comment]
@luomingmeng pls have a look at this pr to make sure it is compatible with the original implementation for reporter

File: pkg/agent/resourcemanager/fetcher/kubelet/kubeletplugin.go (hunk 1 above)
Comment (WangZzzhe) [reply pkg/agent/resourcemanager/fetcher/kubelet/kubeletplugin.go:248]: > will this be necessary to have?

Topology policy needs to be reported in both QRM and ORM modes to ensure that the scheduler can correctly calculate node topology resources

**Review by caohe — DISMISSED** [review]

**Review by caohe — APPROVED** [review]

File: cmd/katalyst-agent/app/options/orm/orm_base.go [hunk 2]
```diff
@@ -58,6 +64,12 @@ func (o *GenericORMPluginOptions) AddFlags(fss *cliflag.NamedFlagSets) {
 		o.TopologyPolicyName, "topology merge policy name used by ORM")
 	fs.StringSliceVar(&o.NumericAlignResources, "numeric-align-resources", o.NumericAlignResources,
 		"resources which should be aligned in numeric topology policy")
+	fs.StringVar(&o.ORMPodResourcesSocket, "orm-pod-resources-socket", o.ORMPodResourcesSocket,
+		"socket of ORM pod resource api, default 'unix:/var/lib/katalyst/pod-resources/kubelet.sock'")
+	fs.StringVar(&o.ORMDevicesProvider, "orm-devices-provider", o.ORMDevicesProvider,
```
Comment (caohe) [inline cmd/katalyst-agent/app/options/orm/orm_base.go:69]: If this config is empty, does it mean that ORM has no requirements for the K8s version?

Comment (WangZzzhe) [reply cmd/katalyst-agent/app/options/orm/orm_base.go:69]: > If the `orm-devices-provider` config is empty, does it mean that ORM has no requirements for the K8s version?

Yes,  there is no other rpc or http connections between ORM and kubelet.

**Review by luomingmeng — APPROVED** [review]

**Pull request merged**